### PR TITLE
CBL-1247: Correct the cookie logic

### PR DESCRIPTION
**Pull request by jayahariv**
* previous parsing logic was wrong, corrected
* include a unit test to validate the parse method
* as per the documentation, https://tools.ietf.org/html/rfc6265#section-4.1.1, any ASCII except control characters can become cookie value, so I have used bell(^G) character

### CODE REVIEW

File: Objective-C/Internal/Replicator/CBLWebSocket.mm [hunk 1]
```diff
@@ -793,4 +784,41 @@ - (void) disconnect {
...
+    NSMutableArray *attrs = [NSMutableArray arrayWithCapacity: [rawAttrs count]];
+    for (NSString* rawAttr in rawAttrs) {
+        // trims the attribute
+        NSString* attr = [rawAttr stringByTrimmingCharactersInSet: [NSCharacterSet whitespaceAndNewlineCharacterSet]];
+        
+        // replace comma with ^G(bell) in Date
+        if ([attr hasPrefix: @"Expires"] || [attr hasPrefix: @"expires"]) {
+            NSRange range = [attr rangeOfString: @","];
+            if (range.location != NSNotFound)
+                attr = [attr stringByReplacingCharactersInRange: range withString: @"^G"];
```
Comment (jayahariv) [inline Objective-C/Internal/Replicator/CBLWebSocket.mm:801]: `^G`

File: Objective-C/Internal/Replicator/CBLWebSocket.mm [hunk 2]
```diff
@@ -793,4 +784,41 @@ - (void) disconnect {
...
+        }
+        [attrs addObject: attr];
+    }
+    
+    // separate out with cookie boundaries
+    NSArray* rawCookies = [[attrs componentsJoinedByString: @";"] componentsSeparatedByString: @","];
+    NSMutableArray *cookies = [NSMutableArray arrayWithCapacity: [rawCookies count]];
+    for (NSString* rawCookie in rawCookies) {
+        // trim the cookie
+        NSString* cookie = [rawCookie stringByTrimmingCharactersInSet: [NSCharacterSet whitespaceAndNewlineCharacterSet]];
```
Comment (pasin) [inline Objective-C/Internal/Replicator/CBLWebSocket.mm:811]: I think you don't need to trim it again, right?

File: Objective-C/Internal/Replicator/CBLWebSocket.mm [hunk 3]
```diff
@@ -793,4 +784,41 @@ - (void) disconnect {
     _out = nil;
 }
 
++ (NSArray*) parseCookie: (NSString*) cookieStr {
```
Comment (pasin) [inline Objective-C/Internal/Replicator/CBLWebSocket.mm:787]: parseCookies: ?

**Review by pasin — APPROVED** [review]
Minor things.

File: Objective-C/Tests/ReplicatorTest+Main.m [hunk 4]
```diff
@@ -1482,6 +1483,21 @@ - (void) testRevisionIdInPushPullFilters {
     Assert([pushDocIds containsObject: @"doc1"]);
 }
 
+- (void) testWebSocketParseCookie {
+    NSArray* inputs = @[
+        @[@"a1=b1", @[@"a1=b1"]],
+        @[@"a1=b1;a2=b2", @[@"a1=b1;a2=b2"]],
+        @[@"a1=b1;expires=b2,3", @[@"a1=b1;expires=b2,3"]],
+        @[@"a1=b1;a2=b2,a3=b3;a4=b4", @[@"a1=b1;a2=b2", @"a3=b3;a4=b4"]],
+        @[@"a1=b1;Expires=Mon, 29 Feb 2021 20:21:12 GMT;Path=/", @[@"a1=b1;Expires=Mon, 29 Feb 2021 20:21:12 GMT;Path=/"]],
```
Comment (pasin) [inline Objective-C/Tests/ReplicatorTest+Main.m:1492]: Another test with two cookies, both have exp (one with E and one with e)?.

**Review by pasin — APPROVED** [review]

File: Objective-C/Internal/Replicator/CBLWebSocket.mm (hunk 2 above)
Comment (jayahariv) [reply Objective-C/Internal/Replicator/CBLWebSocket.mm:811]: `a1=b1,    a2=b2` this space between the cookies are trimmed here.

**Pull request merged**